qa/tasks/fwd_scrub.py

```diff
@@ -0,0 +1,209 @@
+"""
+Thrash mds by simulating failures
+"""
+import logging
+import contextlib
+import threading
+
+from gevent import sleep, killall, joinall, GreenletExit
+from gevent.greenlet import Greenlet
+from gevent.event import Event
+from teuthology import misc as teuthology
+from teuthology import contextutil
+
+from tasks import ceph_manager
+from tasks.cephfs.filesystem import MDSCluster, Filesystem
+from tasks.thrasher import Thrasher
+
+log = logging.getLogger(__name__)
+
+
+class MDSRankScrubber(Thrasher, Greenlet):
+    def __init__(self, fs, mds_rank, sleep_duration=30):
+        super(MDSRankScrubber, self).__init__()
+        self.logger = log.getChild('fs.[{f}]'.format(f=fs.name))
+        self.fs = fs
+        self.mds_rank = mds_rank
+        self.sleep_duration = sleep_duration
+
+    def _run(self):
+        try:
+            self.do_scrub()
+        except Exception as e:
+            self.set_thrasher_exception(e)
+            self.logger.exception("exception:")
+            # allow successful completion so gevent doesn't see an exception...
+
+    def _wait_until_scrub_complete(self, path="/", recursive=True):
+        tag = f"tag-{self.fs.name}-rank-{self.mds_rank}"
+        recopt = ["recursive", "force"] if recursive else []
+        out_json = self.fs.rank_tell(["scrub", "start", path] + recopt + [tag],
+                                     rank=self.mds_rank)
+        assert out_json is not None
+        assert out_json['return_code'] == 0
+        assert out_json['scrub_tag'] == tag
+        assert out_json['mode'] == 'asynchronous'
+
+        duration = self.sleep_duration
+        with contextutil.safe_while(sleep=duration, tries=30) as proceed:
```

Comment:
Sorry I wasn't clear. I think in the interest of ease of use: the duration should be the total duration we want to wait for the scrub to complete. So maybe 5 minutes? Then do:

```suggestion
        with contextutil.safe_while(sleep=total_duration//30, tries=30) as proceed:
```

Reply:
got it; done